File: src/#1457/solution.go
package solution

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func pseudoPalindromicPaths(root *TreeNode) int {
	var res int = 0
	set := [10]int{}
	helper(root, set, &res)
	return res
}

func helper(root *TreeNode, set [10]int, res *int) {
	if root.Left == nil && root.Right == nil {
		set[root.Val]++
		*res += pseudoPalindromicCheck(set)
		return
	}
	set[root.Val]++
	if root.Left != nil {
		helper(root.Left, set, res)
	}
	if root.Right != nil {
		helper(root.Right, set, res)
	}
}

func pseudoPalindromicCheck(set [10]int) int {
	odd := 0
	for _, v := range set {
		if v%2 == 1 {
			odd++
		}
	}
	if odd > 1 {
		return 0
	}
	return 1
}

File: src/#1457/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var pseudoPalindromicPaths = function (root) {
  let res = 0
  let set = new Array(10).fill(0)

  helper(root, set)

  return res

  function helper(root, set) {
    let currentSet = [...set]
    if (root.left === null && root.right === null) {
      currentSet[root.val]++
      res += pseudoPalindromicCheck(currentSet)
      return
    }

    currentSet[root.val]++
    if (root.left !== null) {
      helper(root.left, currentSet)
    }
    if (root.right !== null) {
      helper(root.right, currentSet)
    }
  }
}

function pseudoPalindromicCheck(set) {
  let odd = 0

  set.forEach(v => {
    if (v % 2 === 1) {
      odd++
    }
  })

  if (odd > 1) {
    return 0
  }
  return 1
}
